### PR TITLE
Style fix for EXISTS subquery related changes

### PR DESCRIPTION
**Pull request by jianqiao**
This PR fixes some code style issues for changes in #151, together with minor fixes to class `PatternMatcher` for class `Exists`,

### CODE REVIEW

File: query_optimizer/expressions/SubqueryExpression.cpp [hunk 1]
```diff
@@ -30,7 +30,7 @@ namespace optimizer {
 namespace expressions {
 
 ::quickstep::Scalar* SubqueryExpression::concretize(
-    const std::unordered_map<ExprId, const CatalogAttribute*>& substitution_map) const {
+    const std::unordered_map<ExprId, const CatalogAttribute*> &substitution_map) const {
   LOG(FATAL) << "SubqueryExpression should not be concretized";
   return nullptr;
```
Comment (zuyu) [inline query_optimizer/expressions/SubqueryExpression.cpp]: We could remove this line, as `LOG(FATAL)` terminates the program.

File: query_optimizer/expressions/PatternMatcher.hpp [hunk 2]
```diff
@@ -35,6 +35,7 @@ class BinaryExpression;
 class Cast;
 class ComparisonExpression;
 class Count;
+class Exists;
```
Comment (zuyu) [inline query_optimizer/expressions/PatternMatcher.hpp]: You might want to update the copyright for this file.

**Comment by zuyu** [comment]
LGTM, except for two minor comments. Thanks!

**Comment by jianqiao** [comment]
Updated.

**Comment by zuyu** [comment]
Merged!

**Pull request merged**